delivery_roulier/models/stock_picking.py

```diff
@@ -290,3 +290,17 @@ def open_website_url(self):
         action["domain"] = [("id", "in", packages.ids)]
         action["context"] = {"picking_id": self.id}
         return action
+
+    def _put_in_pack(self, move_line_ids):
```

Comment:
I've splitted correctly the migration commit to `pre-commit` one.
Here is the latest fixup to update `quant_ids` on package with all quants from move lines in this package

Reply:
Hello,
I don't get why we need this, not why it is in this module. Is this necessary for this module to work ? Why ?

Reply:
Hello @florian-dacosta 
From my POV, when the package is created and quants are put in pack, it should be linked to a package.
Probably it was a case in previous versions, but not in v17.
As a result here:
```
def _roulier_get_parcel(self, picking):
        self.ensure_one()
        weight = self.shipping_weight or self.weight
        parcel = {"weight": weight, "reference": self.name}
        return parcel
```
`shipping_weight` or `weight` are missed and roulier response is wrong.
If you can check the code and suggest better solution it would be great.
BTW, I'm not an expert using this module. But, I need it to be migrated for my project.
Thanks in advance.

Reply:
Well, I believe it has always been this way. When you click on "put in pack" button, the pack is set on the operations (stock.move.line) but if you check the pack, it is still empty (no quant inside). This is not specific to the v17.
I understand it may be an issue, if you want to generate the label, before validating the picking, which is not the way odoo works.

The label generation is triggered at the end of "stock.picking._action_done". At this stage, the quant are already linked to the pack, so the weight is ok.

I guess it would be nice to be able to generate label before the validation of the shipping, but : 
- This is not a problem specific to delivery_roulier, I think it could be an issue for any module managing the carrier label. If I should pick a module with something like that, I'd probably create a new dedicated module, which you could choose to install or not, along with delivery_roulier (or any other module managing shipping labels).
- The way to do it does not seem ok to me. Indeed, the button put_in_pack is optional, you can also set the result_package_id of the stock.move.line manually (in this case the weight won't be ok either)
After you set a result package id, you can remove it manually... You can delete the stock.move.line, you can change the pack, you can do anything. Then, the package on some quants will be wrong, which is an issue.
Also, you have a source package and a result package on stock.move.line. When a quant is already in a pack, when you transfer it, the source package is filled. If you write the pack on the quant with the "put in pack" button, then the source package should probably be set, not sure about the impacts.

Anyway, I have no solution to propose around this subject now, but this one should definitely not go here and should probably be well tested as well, with the different cases.

Reply:
Well, I believe it has always been this way. When you click on "put in pack" button, the pack is set on the operations (stock.move.line) but if you check the pack, it is still empty (no quant inside). This is not specific to the v17.
I understand it may be an issue, if you want to generate the label, before validating the picking, which is not the way odoo works.

The label generation is triggered at the end of "stock.picking._action_done". At this stage, the quant are already linked to the pack, so the weight is ok.

I guess it would be nice to be able to generate label before the validation of the shipping, but : 
- This is not a problem specific to delivery_roulier, I think it could be an issue for any module managing the carrier label. If I should pick a module with something like that, I'd probably create a new dedicated module, which you could choose to install or not, along with delivery_roulier (or any other module managing shipping labels).
- The way to do it does not seem ok to me. Indeed, the button put_in_pack is optional, you can also set the result_package_id of the stock.move.line manually (in this case the weight won't be ok either)
After you set a result package id, you can remove it manually... You can delete the stock.move.line, you can change the pack, you can do anything. Then, the package on some quants will be wrong, which is an issue.
Also, you have a source package and a result package on stock.move.line. When a quant is already in a pack, when you transfer it, the source package is filled. If you write the pack on the quant with the "put in pack" button, then the source package should probably be set, not sure about the impacts.

Anyway, I have no solution to propose around this subject now, but this one should definitely not go here and should probably be well tested as well, with the different cases.